### PR TITLE
Catalog for alter table and change column name.

### PR DESCRIPTION
**Pull request by DeanChensj**

### CODE REVIEW

File: src/catalog/catalog.cpp [hunk 1]
```diff
@@ -787,6 +793,253 @@ std::shared_ptr<TableCatalogObject> Catalog::GetTableObject(
...
+    columnId = schema->GetColumnID(old_names[0]);
+    if (columnId == INVALID_OID) {
+      throw CatalogException("Old column does not exist in the table.");
+    }
+
+    // Change column name in the global schema
+    schema->ChangeColumnName(columnId, names[0]);
+
+    // TODO: verify do we really need doing so
+    // Modify all the tiles within the Data Table
```
Comment (Dingshilun) [inline src/catalog/catalog.cpp]: shall we delete those commented code? Anyway we won't use them as the correct way to change name has been figured out

File: src/include/storage/tile.h [hunk 2]
```diff
@@ -139,6 +139,11 @@ class Tile : public Printable {
     return schema.GetColumn(column_index).GetName();
   }
 
+  inline void ChangeColumnName(const oid_t column_index,
```
Comment (Dingshilun) [inline src/include/storage/tile.h]: do we need to change column names in tile?

File: src/catalog/catalog.cpp (hunk 1 above)
Comment (DeanChensj) [reply src/catalog/catalog.cpp]: For now it works fine in simple psql test, but I'm not sure if it won't cause inconsistency problem...

File: src/include/storage/tile.h (hunk 2 above)
Comment (DeanChensj) [reply src/include/storage/tile.h]: See above.

**Pull request merged**